tests/autofix_files/async100.py

```diff
@@ -114,3 +114,9 @@ async def more_nested_tests():
     with contextlib.suppress(Exception):
         with open("blah") as file:
             print("foo")
+
+
+# Don't trigger for blocks with a yield statement
+async def foo():
+    with trio.fail_after(1):
+        yield
```

Comment:
Ah.  This should indeed be allowed by `ASYNC100`...

...but now I'm wondering about a non-optional rule to warn when you yield across a known cancelscope and the function isn't a context-manager, [because that really doesn't work](https://discuss.python.org/t/preventing-yield-inside-certain-context-managers/1091).  Probably not worth the trouble, since njs and I are planning to get runtime enforcement, the rule would have many missed alarms, and `ASYNC900` covers it anyway for people who care.